### DISCUSSION TITLE
How to capitilize first char of `ref` `supplement` if used at the sentence beginning?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2009] [Q&A]
I've used `#set text(lang: "ru")`, but even though the default `supplement`s are OK, my requirements _require_ me to change them:

```typ
#show ref: it => {
  let fig = it.element
  if fig == none or fig.func() != figure { return it }
  let in_fig = fig.body.func()
  let number = numbering(fig.numbering, counter(figure).at(fig.location()).first())
  let prefix
  if in_fig == image { prefix = "рис" } // For figure(image())
  else if in_fig == table { prefix = "табл" } // For figure(table())
  else { return it }
  link(fig.location())[#prefix. #number]
}
```

<details><summary>same with figure</summary>

But since the only thing that can't be removed from `figure` is a `:` (!), I have to reconstruct everything from scratch (instead of much simpler `set`/`show` setting):

```typ
#show figure.where(kind: image): it => align(center)[
  #it.body
  Рис. #it.counter.display(). #it.caption
]
#show figure.where(kind: table): it => align(center)[
  #align(right)[Таблица #it.counter.display(). #it.caption]
  #it.body
]
```

</details>

But this means that I can't use capital letter if I start my sentence with the label reference. In [hayagriva](https://github.com/typst/hayagriva/blob/main/docs/file-format.md#formattable-string) format there is something called `title-case` and `sentence-case` fields for strings, which probably the same thing, that I'm trying to solve. Is there any way to resolve this issue?

P.S. Even `#show regex("^рис."): "Рис."` doesn't work! It ignores `^` (and `\A`). And I can't use spaces (they don't work) in the regex, so I can't even apply the whole pattern (with spaces).

**Comment by gzagatti** [comment]
I have the same problem. Did you find a solution?

**Reply by Andrew15-5** [reply]
No, I think it's too hard to implement this in Typst right now. Maybe something like https://typst.app/christmas/4 is possible to do, but I haven't looked closely at the logic, so I can't comment on this.

**Reply by gzagatti** [reply]
I just saw the following in the [documentation](https://typst.app/docs/reference/model/ref/), so I can just customize my supplement. It should be enough for me.

> To customize the supplement, add content in square brackets after the reference: @intro[Chapter].

**Reply by Andrew15-5** [reply]
Yes, this is the easy workaround. I don't like that I would have to do this the explicit way. The cleaner the file the better. But if it is absolutely necessary, then the `@label[supplement]` syntax is available.
